inverse.go

```diff
@@ -184,16 +184,56 @@ func (i *InverseBloomFilter) WriteTo(stream io.Writer) (int64, error) {
 // have been written by WriteTo()) from an i/o stream. It returns the number
 // of bytes read.
 func (i *InverseBloomFilter) ReadFrom(stream io.Reader) (int64, error) {
+	decoded, capacity, err := i.decodeToArray(stream)
+	if err != nil {
+		return int64(0), err
+	}
+
+	// Create []*[]byte and point to each item in decoded
+	decodedWithPointers := make([]*[]byte, capacity)
+	for p := range decodedWithPointers {
+		if len(decoded[p]) == 0 {
+			decodedWithPointers[p] = nil
+		} else {
+			decodedWithPointers[p] = &decoded[p]
+		}
+	}
+
+	i.array = decodedWithPointers
+	i.capacity = uint(capacity)
+	return int64(capacity) + int64(2*binary.Size(uint64(0))), nil
+}
+
+// ImportFrom reads a binary representation of InverseBloomFilter (such as might
+// have been written by WriteTo()) from an i/o stream. It returns the length of
+// the received elements.
+func (i *InverseBloomFilter) ImportElementsFrom(stream io.Reader) (int, error) {
+	decoded, _, err := i.decodeToArray(stream)
+	if err != nil {
+		return 0, err
+	}
+
+	// Create []*[]byte and point to each item in decoded
+	for p := range decoded {
+		if len(decoded[p]) > 0 {
+			i.Add(decoded[p])
+		}
+	}
+
+	return len(decoded), nil
+}
+
+func (i *InverseBloomFilter) decodeToArray(stream io.Reader) ([][]byte, uint64, error){
```

Comment:
It was originally inline with `ReadFrom` but I broke it into a helper method to support `ImportElementsFrom` as well